### PR TITLE
Experimental support for GISTIC export

### LINKED ISSUE #622: How to export gistic markers?
Within `export.py` there's a `export_gistic_markers(cnr_fnames)`.
Also within `export.py`, there's a dictionary...
```
EXPORT_FORMATS = {
    'cdt': fmt_cdt,
    # 'gct': fmt_gct,
    'gistic': export_gistic_markers,
    'jtv': fmt_jtv,
    'nexus-basic': export_nexus_basic,
    'nexus-ogt': export_nexus_ogt,
    'seg': export_seg,
    'theta': export_theta,
    'vcf': export_vcf,
}
```

I've tried exporting a gistic marker file using `cnvkit.py export gistic` but I get an error...
```
# cnvkit.py export gistic
usage: cnvkit.py export [-h] {bed,seg,vcf,theta,nexus-basic,nexus-ogt,cdt,jtv} ...
cnvkit.py export: error: invalid choice: 'gistic' (choose from 'bed', 'seg', 'vcf', 'theta', 'nexus-basic', 'nexus-ogt', 'cdt', 'jtv')
```

**What is the usage to get a gistic marker output file from cnvkit?**

### PR DESCRIPTION
**Pull request by tetedange13**
Should fix issue #622 
=> Simply copy-pasted (+ adapted) code for 'jtv' export (same case with multiple ".cnr" as input) to enable 'gistic' export through command-line
=> This supposes "core" function for "gistic" export (within `cnvlib/export.py`) is correct, which I have not test
=> Maybe further tests are required ?

Closes #622

### CODE REVIEW

**Comment by etal** [comment]
Looks good, thanks. I think I left this feature as hidden because I hadn't tested its results with Gistic myself. Are you in a position to check whether the output of this command works with Gistic without further modification?

**Comment by tetedange13** [comment]
Hi @etal ,
<br>

Sorry, I have never used GISTIC and I am completely unfamiliar with it
=> However I installed it and as far as I understood it takes 2 input files that can be both produced by `cnvkit.py export` 
- A mandatory "seg" file, whose format matches output from `cnvkit.py export seg <SEVERAL_CNS>` (I mean GISTIC is taking it without error)
- An optionnal "markers" file which is produced by `cnvkit.py export gisitic <SEVERAL_CNR>`

Feeding GISTIC **only** with mandatory "seg" file works fine, but I got an **issue feeding it with both "seg" and "markers" files**:
```bash
GISTIC 2.0 input error detected:
74 segment start or end positions in 'cns.gistic.seg' do not match any markers in 'cnr.gistic.markers'.
First bad position is X:150501 at line 114.
```
=> I guess it is because `cnvkit.py export gistic` [works only on autosomes](https://github.com/etal/cnvkit/blob/25d91cdb962639c98d041c3b05c7de659c41dc70/cnvlib/export.py#L372) (contrary to `cnvkit.py export seg`)
=> Fortunately, filtering out sexual chromosomes from "seg" file produced by CNVkit is enough for GISTIC to finish without error !

To conclude: this seems OK to me, but as I said I do not know enough about GISTIC to assert results generated from "GISTIC + CNVkit's exports" are accurate
=> Maybe you should rather wait for `BioComSoftware` feedback ?
<br>

 Hope this helps.
Have a nice day.
Felix !

**Comment by tskir** [comment]
@tetedange13 Thank you for the PR Felix! I agree we should ideally wait for @BioComSoftware's feedback prior to merging this.

I also think there are two additional things to be done here:
* Adding some basic documentation about this format to the export section;
* Since GISTIC presumably can only work with autosomes, the SEG export command should have a flag to only export autosomes, to avoid the mismatch which you encountered in your tests.

**Comment by tetedange13** [comment]
Hi @tskir,
<br>

> Since GISTIC presumably can only work with autosomes

Just to be clear, GISTIC itself can deal with sex-chrom (even if they are removed by default according to [doc](http://portals.broadinstitute.org/cgi-bin/cancer/publications/pub_paper.cgi?mode=view&paper_id=216&p=t) and that I could not manage to make it use keep them with `-rx=0` param...)
=> The issue I was getting comes from inconsistency between `cnvkit.py export seg` that includes sex-chrom and current `cnvkit.py export gistic` that removes them internally (based on [here](https://github.com/etal/cnvkit/blob/25d91cdb962639c98d041c3b05c7de659c41dc70/cnvlib/export.py#L372))
=> I do not know if there was a specific reason for that, at the time `export_gistic_markers()` was written?

To sum up:
=> If you want I can commit a sex-chrom switch for `cnvkit.py export seg` 
=> But would not it be better to simply remove the `.autosome()` filter from `export_gistic_markers()` ? (I tested, GISTIC is taking both files just fine)

<br>

Kind regards.
Felix.

**Comment by tskir** [comment]
Based on the feedback we received, I think this would be the most reasonable set of changes:
* Remove the `.autosome()` filter
* Add minimal description of gistic export to the [documentation](https://github.com/etal/cnvkit/blob/master/doc/importexport.rst) and mark it as experimental

@tetedange13 Do you think you could add those changes to this PR?

**Comment by tetedange13** [comment]
Hi @tskir ,

Done ! To document GISTIC export I did my best based on code docstrings + GISTIC documentation

Have a nice day.
Felix.

**Review by tskir — APPROVED** [review]
@tetedange13 Thank you Felix, you're a star!

**Review by etal — APPROVED** [review]
LGTM, thanks! The `.autosomes()` filter was there because I was under the mistaken impression that GISTIC only accepts autosomes as input.
